lib/client-assets.php

```diff
@@ -1189,3 +1189,51 @@ function gutenberg_editor_scripts_and_styles( $hook ) {
 	 */
 	do_action( 'enqueue_block_editor_assets' );
 }
+
+/**
+ * Ensure the editor module is loaded before third party plugins.
+ *
+ * Remove this in Gutenberg 3.1
+ */
+function polyfill_blocks_module_in_scripts() {
+	if ( ! is_gutenberg_page() ) {
+		return;
+	}
+
+	wp_enqueue_script( 'wp-editor' );
+}
+
+add_action( 'enqueue_block_editor_assets', 'polyfill_blocks_module_in_scripts', 9 );
+add_action( 'enqueue_block_assets', 'polyfill_blocks_module_in_scripts', 9 );
+
+
+/**
+ * This helper function ensures, that every item in $color_palette has a slug.
+ *
+ * @access public
+ * @param array $color_palette The color palette registered with theme_support.
+ * @return array $new_color_palette The color palette with slugs added where needed
+ */
+function editor_color_palette_slugs( $color_palette ) {
+	$new_color_palette = array();
+	$is_doing_it_wrong = false;
+
+	foreach ( $color_palette as $color ) {
+		if ( ! isset( $color['slug'] ) ) {
+			$color['slug']     = esc_js( $color['name'] );
+			$is_doing_it_wrong = true;
+		}
+
+		$new_color_palette[] = $color;
+	}
+
+	if ( $is_doing_it_wrong ) {
+		_doing_it_wrong(
+			'add_theme_support/editor-color-palette',
```

Comment:
This needs to be the function name only, i.e. add_theme_support(). I agree though that it makes sense to highlight the theme feature, so maybe in the description you could say "Each color specified for the editor color palette...".